### PR TITLE
Add example accelerator using HLS

### PR DESCRIPTION
**Pull request by schwarz-em**
**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [x] New feature
- [ ] Other enhancement

**Impact**:
- [x] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [ ] Other

### CODE REVIEW

File: .github/workflows/chipyard-run-tests.yml
Comment (abejgonzalez) [inline .github/workflows/chipyard-run-tests.yml:1]: Since you are just building you don't need this (you aren't running tests)

File: docs/Customization/Incorporating-HLS.rst
Comment (abejgonzalez) [inline docs/Customization/Incorporating-HLS.rst:14]: I would use `gh-file-ref` to have it auto-generate a link: https://github.com/ucb-bar/chipyard/blob/0ad87ff5cec66396f3b0de4d639de1b2ef7221c5/docs/Simulation/FPGA-Accelerated-Simulation.rst#L26

File: docs/Customization/Incorporating-HLS.rst
Comment (abejgonzalez) [inline docs/Customization/Incorporating-HLS.rst:19]: The CI is using an older version of the tool FYI.

File: docs/Customization/Incorporating-HLS.rst
Comment (abejgonzalez) [inline docs/Customization/Incorporating-HLS.rst:29]: All code block directives need an empty space before the main text of the block.

File: docs/Customization/Incorporating-HLS.rst
Comment (abejgonzalez) [inline docs/Customization/Incorporating-HLS.rst:35]: Below you mention this also does the "move files to vsrc" step. So maybe I would add some extra saying "to script running HLS (as well as copying files around as explained later)"

File: generators/hls-example/accel/HLSAccel.hpp
Comment (abejgonzalez) [inline generators/hls-example/accel/HLSAccel.hpp:1]: Nitpicking a bit, can we just combine the HPP into the CPP file (instead of having two files).

File: generators/hls-example/accel/HLSAccel.cpp
Comment (abejgonzalez) [inline generators/hls-example/accel/HLSAccel.cpp:1]: Indentation is off in this file.

File: docs/Customization/Incorporating-HLS.rst
Comment (abejgonzalez) [reply docs/Customization/Incorporating-HLS.rst:19]: Also maybe I would point to some docs/link to Vitis HLS. Isn't there restrictions on where this can be used (and if so, should we mention something about that here)?

File: generators/chipyard/src/main/scala/DigitalTop.scala
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/DigitalTop.scala:1]: Can we change the naming to something like `CanHavePeripheryHLSGCD`. Ditto for the config. fragment name.

File: docs/Customization/Incorporating-HLS.rst
Comment (abejgonzalez) [inline docs/Customization/Incorporating-HLS.rst:1]: Can you also mention the TODO of how to connect up to an AXI interface.

File: generators/hls-example/run_hls.tcl
Comment (abejgonzalez) [inline generators/hls-example/run_hls.tcl:2]: HPP is missing?

File: generators/hls-example/run_hls.tcl
Comment (abejgonzalez) [inline generators/hls-example/run_hls.tcl:5]: Why do you need a part number? This is FPGA specific so maybe the docs should mention this RTL that is made is specific to an FPGA (with part number listed here) but can be run in RTL sim.

File: generators/hls-example/run_hls.tcl
Comment (abejgonzalez) [inline generators/hls-example/run_hls.tcl:6]: I see that both the FPGA part number and the clock period should probably configurable from the makefile and piped to the Chisel.

**Review by abejgonzalez — CHANGES_REQUESTED** [review]

**Comment by abejgonzalez** [comment]
Super super cool! Thanks Ella!

**Review by abejgonzalez — COMMENTED** [review]
Also, I think it might be good to combine this into the existing GCD example area instead of separate SBT project s.t. people can either create a GCD with 1. Chisel 2. SV 3. SV from HLS

File: build.sbt
Comment (jerryz123) [inline build.sbt:266]: I think this should go in generators/chipyard/example, instead of its own project

File: docs/Customization/Incorporating-HLS.rst
Comment (schwarz-em) [reply docs/Customization/Incorporating-HLS.rst:19]: I looked closer on the product site and the documentation but I couldn't find any restrictions on where Vitis HLS can be used. Possibly it's in the license somewhere, but I don't really want to crawl through that... Some forum comments implied that Vitis HLS has been used in commercial tapeouts, so if there are rules they don't seem to be very strict. We should be able to just leave it up to the user to know what they're doing.

File: generators/chipyard/src/main/scala/example/GCD.scala
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/example/GCD.scala:339]: Can you just reuse the above fragment and have a check to ensure only one flag is enabled at a time.

**Review by abejgonzalez — CHANGES_REQUESTED** [review]
Small nits otherwise LGTM

**Review by abejgonzalez — COMMENTED** [review]
LGTM pending CI

**Review by jerryz123 — APPROVED** [review]

**Review by abejgonzalez — APPROVED** [review]

**Pull request merged**
